zenoh-flow/src/runtime/loops.rs

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright (c) 2017, 2021 ADLINK Technology Inc.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ADLINK zenoh team, <zenoh@adlink-labs.tech>
+//
+
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use uhlc::Timestamp;
+
+use crate::{model::loops::LoopDescriptor, NodeId};
+
+/// A `LoopContext` is associated to each message that goes through a Loop.
+///
+/// It gives the following information:
+/// - the ingress / egress,
+/// - an iteration counter (if the Loop is finite),
+/// - the starting timestamp of the first iteration,
+/// - the starting timestamp of the current iteration,
+/// - the duration of the last iteration (after the first full iteration is performed).
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LoopContext {
+    pub(crate) ingress: NodeId,
+    pub(crate) egress: NodeId,
+    pub(crate) iteration: LoopIteration,
+    pub(crate) timestamp_start_first_iteration: Timestamp,
+    pub(crate) timestamp_start_current_iteration: Option<Timestamp>,
+    pub(crate) duration_last_iteration: Option<Duration>,
+}
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub enum LoopIteration {
+    Finite(usize),
```

Comment:
Use `u64` instead of `usize`, to have the same size also across different machines/architectures.